src/osmf/lib/HDSPlayer.as

```diff
@@ -834,7 +839,15 @@ package
     public function onGetCurrentTime(event:Event):void
     {
       var eventObject:Object = new Object();
-      eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.currentTime.toString();
+      if (_seekToEnd == true)
+      {
+        eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.duration.toString();
+        _seekToEnd = false;
```

Comment:
Hi Angie,

As per the new approach, the situation of seeking to the end of the video
is detected inside the onVideoSeek() function itself. Accordingly changes
are done inside getCurrentTime(), to handle the situation. The dependency
of the logic with onPlayCompleted() is removed.


Thanks

Sarita Singh




On Tue, Apr 26, 2016 at 11:39 AM, angieisrani <notifications@github.com>
wrote:

> In src/osmf/lib/HDSPlayer.as
> <https://github.com/ooyala/html5-video-plugins/pull/144#discussion_r61141019>
> :
>
> > @@ -834,7 +839,15 @@ package
> >      public function onGetCurrentTime(event:Event):void
> >      {
> >        var eventObject:Object = new Object();
> > -      eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.currentTime.toString();
> > +      if (_seekToEnd == true)
> > +      {
> > +        eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.duration.toString();
> > +        _seekToEnd = false;
>
> If I could call getCurrentTime at any time, what does getCurrentTime have
> to do with seekToEnd? Why does it get reset to false here? Shouldn't it
> remain false until we are actually not at the end anymore?
>
> Also this variable is now specific to seeking but what if I call
> getCurrentTime when you didn't seek? The behaviour of getCurrentTime
> shouldn't depend on how you go to the end of the stream; just if you got
> there
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/ooyala/html5-video-plugins/pull/144/files/4e0533e1ef55d57fd89a661b40e27a361b9e5a3a#r61141019>
>



-- 
Best Regards,
Sarita Singh